src/main/java/hu/blackbelt/structured/map/proxy/MapProxy.java

```diff
@@ -706,6 +708,82 @@ private boolean invokeEquals(Object proxy, Object[] args) throws InvocationTarge
         return false;
     }
 
+    public void invokeAddOrRemove(Method m, Object[] args, boolean add) throws ExecutionException {
```

Comment:
This method has to split into more parts. Method / blocks with names make it easier to understand the code. Also helps debugging.

Reply:
This is part of my previous PR. I originally created separate functions for addto and removefrom, but there are minimal differences between addto and removefrom, so @robertcsakany asked to merge the two.